### PR TITLE
Adding Page, Web, Team and User contexts

### PR DESCRIPTION
**Pull request by lohitakshgupta**
This PR adds the following contexts:

1. Page
2. Web
3. Team
4. User 

Changes come from internal resources and the following PRs [Note: Not all changes have been ported]:
#37 
#49

### CODE REVIEW

File: src/SDK.ts [hunk 1]
```diff
@@ -135,6 +104,275 @@ export interface IExtensionContext {
...
+}
+
+interface MailSettings {
+    enabled: boolean;
+}
+
+/**
+* Web Access configuration data. This information is used to process requests on the server.  This data is also placed in a json island on each page in order for JavaScript to know key configuration data required to things like construct proper urls
+*/
+interface ConfigurationContext {
```
Comment (mihaelams1) [inline src/SDK.ts:290]: Consider renaming it to EnvironmentContext

File: src/SDK.ts [hunk 2]
```diff
@@ -135,6 +104,275 @@ export interface IExtensionContext {
...
+interface TimeZonesConfiguration {
+    daylightSavingsAdjustments: DaylightSavingsAdjustmentEntry[];
+}
+
+interface MailSettings {
+    enabled: boolean;
+}
+
+/**
+* Web Access configuration data. This information is used to process requests on the server.  This data is also placed in a json island on each page in order for JavaScript to know key configuration data required to things like construct proper urls
```
Comment (mihaelams1) [inline src/SDK.ts:288]: Not actual anymore. "Environment configuration data" as a suggestion

File: src/SDK.ts [hunk 3]
```diff
@@ -135,6 +104,275 @@ export interface IExtensionContext {
...
+*/
+interface ConfigurationContext {
+    isHosted: boolean;
```
Comment (mihaelams1) [inline src/SDK.ts:291]: Let's add a comment, something along the lines of "Distinguish between Azure DevOps Services (true) and Azure DevOps Server (false)"

File: src/SDK.ts [hunk 4]
```diff
@@ -135,6 +104,275 @@ export interface IExtensionContext {
...
+}
+
+/**
+* Global context placed on each web page (through json island data) which gives enough information for core TypeScript modules/controls on the page to operate
```
Comment (mihaelams1) [inline src/SDK.ts:295]: The comment doesn't apply to extensions. "Global context placed on each web page" might suffice

File: src/SDK.ts [hunk 5]
```diff
@@ -135,6 +104,275 @@ export interface IExtensionContext {
...
+    /**
+    * Current navigation context.
+    */
+    navigation: NavigationContext;
+    /**
+    * Contains global time zone configuration information (e.g. which dates DST changes)
+    */
+    timeZonesConfiguration: TimeZonesConfiguration;
+    /**
+    * Web Access configuration
```
Comment (mihaelams1) [inline src/SDK.ts:316]: Update as also suggested on the interface

**Review by mihaelams1 — CHANGES_REQUESTED** [review]

File: src/SDK.ts [hunk 6]
```diff
@@ -135,6 +104,275 @@ export interface IExtensionContext {
...
+    * Date that the offset adjustment starts
+    */
+    start: Date;
+}
+
+interface TimeZonesConfiguration {
+    daylightSavingsAdjustments: DaylightSavingsAdjustmentEntry[];
+}
+
+interface MailSettings {
```
[the remark this answers is not in the document]
Comment (lohitakshgupta) [reply src/SDK.ts:283]: Removed

File: src/SDK.ts [hunk 7]
```diff
@@ -135,6 +104,275 @@ export interface IExtensionContext {
...
+*/
+export interface IPageContext {
+    /**
+    * Globalization data for the current page based on the current user's settings
+    */
+    globalization: GlobalizationContext;
+    /**
+    * Cached set of hubs and hub groups for the given request/navigation-context
+    */
+    hubsContext: HubsContext;
```
[the remark this answers is not in the document]
Comment (lohitakshgupta) [reply src/SDK.ts]: Resolving it as content is removed. Now, there's a 1:1 reflection with the payload being returned by the SDK provider.

File: src/SDK.ts [hunk 8]
```diff
@@ -135,6 +104,275 @@ export interface IExtensionContext {
...
+
+    /**
+    * Current navigation context.
+    */
+    navigation: NavigationContext;
```
[the remark this answers is not in the document]
Comment (lohitakshgupta) [reply src/SDK.ts]: Resolving it as content is removed. Now, there's a 1:1 reflection with the payload being returned by the SDK provider.

File: src/SDK.ts (hunk 1 above)
Comment (lohitakshgupta) [reply src/SDK.ts]: Resolving it as content is removed. Now, there's a 1:1 reflection with the payload being returned by the SDK provider.

File: src/SDK.ts (hunk 3 above)
Comment (lohitakshgupta) [reply src/SDK.ts]: `isHosted` is present under the `IHostContext` interface.
Resolving it as content is removed. Now, there's a 1:1 reflection with the payload being returned by the SDK provider.

File: src/SDK.ts (hunk 2 above)
Comment (lohitakshgupta) [reply src/SDK.ts]: Interface removed already

File: src/SDK.ts (hunk 4 above)
Comment (lohitakshgupta) [reply src/SDK.ts:176]: Done

**Review by mihaelams1 — APPROVED** [review]

**Pull request merged**